radixdlt-core/radixdlt/src/integration/java/com/radixdlt/harness/MockedPeersViewModule.java

```diff
@@ -88,10 +89,9 @@ public PeersView peersView(@Self BFTNode self, ImmutableList<BFTNode> nodes) {
   }
 
   private ImmutableList<BFTNode> filterNodes(BFTNode self, ImmutableList<BFTNode> allNodes) {
```

Comment:
Seems we don't need `allNodes`, we only return it. It also seems a bit strange that we are passing a list of nodes to a filter method, but it is using a private nodes list in its logic. Shouldn't we be returning the private list?

Reply:
It returns a predefined list of peers for the given node, if it's defined in the `nodes` map. If for the given node there's no entry in the map, its PeersView contains all nodes in the network (the allNodes) parameter.
The name `filterNodes` seems quite wrong. I've refactored this class, should be much clearer now.

Reply:
I'm not sure the `allNodes` parameter is necessary. We can just return an `Optional` and let the client decide how to handle the case when there's no entry in the map.